extern/CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+# ========================================================================================
+# File  : extern/CMakeLists.txt
+# Brief : CMake file controlling retrieval and compilation of external dependencies
+# ========================================================================================
+include(FetchContent)
+
+# ----------------------------------------------------------------------------------------
+# Abseil
+# ----------------------------------------------------------------------------------------
+FetchContent_Declare(
+    abseil
```

Comment:
Is this just for general use across the project, or do you have something specific in mind?

Reply:
Googletest can use it and it offers a better hash map implementation. I can remove it, though.

Reply:
Works for me - was mostly just curious what you had in mind.